Parse/Internal/Object/Subclassing/ObjectSubclassingController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+
+#if UNITY
+using TypeInfo = System.Type;
+#endif
+
+namespace Parse.Internal {
+  internal class ObjectSubclassingController : IObjectSubclassingController {
+    private ReaderWriterLockSlim mutex;

```

Comment:
@hallucinogen that's really the only reason. I figure this will be a high-throughput critical section, and for most accesses there's no need to be exclusive.